mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosCalls.scala

```diff
@@ -6,7 +6,7 @@ import org.apache.mesos.v1.Protos.{AgentID, ExecutorID, Filters, FrameworkID, Ki
 import org.apache.mesos.v1.scheduler.Protos.Call
 import org.apache.mesos.v1.scheduler.Protos.Call.{Accept, Decline, Reconcile, Revive, UpdateFramework}
 
-class MesosCalls(frameworkId: FrameworkID) {
+class MesosCalls() {
```

Comment:
This is a nice untangling IMHO. Since the client know the framework ID it can just inject int via the Mesos sink.

Reply:
The frameworkId won't change though, right? Isn't it still sufficient to reuse the mesosCalls factory emitted from the first materialization?

Reply:
The framework id should not change. And we should not use the factory. As you see above it would make the scheduler logic depend on the factory from the client instead of just the flow.